newrelic-alerts-configurator/src/test/java/com/ocadotechnology/newrelic/alertsconfigurator/ConditionConfiguratorTest.java

```diff
@@ -90,6 +90,23 @@ public void shouldDoNothing_whenNoChannelsInConfiguration() {
         // when
         testee.sync(config);
 
+        // then
+        InOrder order = inOrder(alertsConditionsApiMock);
+        order.verifyNoMoreInteractions();
+    }
+
+    @Test
+    public void shouldDoNothing_whenEmptyConditionsInConfiguration() {
+        // given
+        PolicyConfiguration config = PolicyConfiguration.builder()
+                .policyName(POLICY_NAME)
+                .incidentPreference(PolicyConfiguration.IncidentPreference.PER_POLICY)
+                .conditions(Collections.emptyList())
+                .build();
+
+        // when
+        testee.sync(config);
+
         // then
         InOrder order = inOrder(alertsConditionsApiMock);
         order.verify(alertsConditionsApiMock).list(POLICY.getId());
```

Comment:
It's not exactly nothing. Something happens. However this test should rather verify that in case of empty conditions list, all existing conditions are removed, not that list of conditions is retrieved from the server. Retrieving the list does not mean that any condition will be removed. This test is too tightly coupled with the internal implementation.
This applies to other tests classes for different condition types.